content/sensu-go/5.17/reference/secrets.md

```diff
@@ -0,0 +1,218 @@
+  - [Top-level attributes](#top-level-attributes) | [Metadata attributes](#metadata-attributes) | [Spec attributes](#spec-attributes)
+- [Secret configuration](#secret-configuration)
+- [Secret examples](#secret-examples)
+
+**COMMERCIAL FEATURE**: Access the Secret datatype in the packaged Sensu Go distribution.
+For more information, see [Get started with commercial features][1].
+
+Sensu's secrets management eliminates the need to expose secrets in your Sensu configuration.
+When a Sensu resource definition requires a secret (e.g. a username or password), Sensu allows you to obtain secrets from one or more external secrets providers, so you can both refer to external secrets and consume secrets via backend [environment variables][5].
+
+Only Sensu backends have access to request secrets from a [secret provider][7].
+Sensu backends cache fetched secrets in memory, with no persistence to a Sensu datastore or file on disk.
+Secrets provided via a "lease" with a "lease duration" are deleted from Sensu's in-memory cache after the configured number of seconds, prompting the Sensu backend to request the secret again.
+
+Secrets are only transmitted over a TLS websocket connection.
+Unencrypted connections must not transmit privileged information.
+For agent-side resources, enable TLS/mTLS.
+
+Secrets are only exposed to Sensu services like environment variables and are automatically redacted from all logs, the API, and the Sensu dashboard.
+ 
+## Secret specification
+
+### Top-level attributes
+
+type         | 
+-------------|------
+description  | Top-level attribute that specifies the resource type. For secrets configuration, the type should always be `Secret`.
+required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
+type         | String
+example      | {{< highlight shell >}}"type": "Secret"{{< /highlight >}}
+
+api_version  | 
+-------------|------
+description  | Top-level attribute that specifies the Sensu API group and version. For secrets configuration in this version of Sensu, the `api_version` should always be `secrets/v1`.
+required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
+type         | String
+example      | {{< highlight shell >}}"api_version": "secrets/v1"{{< /highlight >}}
+
+metadata     |      |
+-------------|------
+description  | Top-level scope that contains the secret's `name` and `namespace`.
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"metadata": {
+  "name": "sensu-ansible-token",
+  "namespace": "default"
+}
+{{< /highlight >}}
+
+spec         | 
+-------------|------
+description  | Top-level map that includes secrets configuration [spec attributes][8].
+required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"spec": {
+  "id": "ANSIBLE_TOKEN",
```

Comment:
Would this actually need to be `SENSU_ANSIBLE_TOKEN` to follow config instructions at https://docs.sensu.io/sensu-go/5.17/reference/backend/#configuration-via-environment-variables?

Reply:
No, the `SENSU_` prefix is only needed for environment variables that are used as backend configuration options, ex. `SENSU_API_LISTEN_ADDRESS=192.168.100.20:8080` instead of `--api-listen-address 192.168.100.20:8080`.